### ISSUE TITLE
Setting NOUP during a cluster reboot results in broken OSDs in CrashLoopBackOff, due to too short livenessProbe.periodSeconds 

### ISSUE THREAD

**Issue by psarossy** [rook/rook#5492]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
* I had to move the cluster physically, which required a power down.
* After turning the cluster back up, several OSDs are in constant CrashLoopBackOff without generating any logs (we'll get to why).

**Expected behavior:**
The cluster to come back without errors.

**How to reproduce it (minimal and precise):**
* I had a healthy cluster before the migration
* Set **noup**,nodown,noout,nobackfill,norecover,noscrub,nodeep-scrub on the cluster (based on another website as the Rook wiki has no info on this)
* Waited a few minutes.
* Powered down the nodes one by one.
* Did the physical move
* Powered the nodes back on (starting with the kubernetes master)
* Waited a few minutes
* Ran ceph -s to see that the cluster is not healthy.

Cannot get any logs out of the OSD pod.
It only creates a single line:
`2020-05-16 18:02:30.394 7f06d8e79a80 -1 Falling back to public interface`

This is all that exists in the kubernetes or docker logs.

In some guides, for cluster reboots they also recommend for you to set "NOUP" on the OSDs.

Naturally this prevents the OSD from starting up properly:
```
$ kubectl logs -f -n rook-ceph rook-ceph-osd-14-cb8958458-dvfzc
2020-05-17 21:32:33.954 7fd5f721ea80 -1 Falling back to public interface
2020-05-17 21:33:50.267 7fd5f721ea80 -1 osd.14 21144 log_to_monitors {default=true}
2020-05-17 21:33:50.363 7fd5dd007700 -1 osd.14 21144 osdmap NOUP flag is set, waiting for it to clear
```

The problem is that the liveliness check only gives 10 seconds for the logs to get populated, but it takes ~30 seconds for that error message to show up, so the logs will not tell you this.

I ended up editing the deployment (stopped/scaled the operator first):
`$ kubectl -n rook-ceph edit deployment rook-ceph-osd-14`

And setting the livenessProbe to periodSeconds to 600 (but anything more than 1 minute would work), then when either just watch the OSD figure out that the NOUP has cleared on it's own or issue an `ceph osd unset noup` then delete/restart the pod, and "surprise", everything is back to normal.

**Essentially it takes up to a minute to get a new OSDMAP and flags from the monitors, the config only gives them 10 seconds and kills them actively.**

**File(s) to submit**:

* Cluster CR: [cluster.yaml.log](https://github.com/rook/rook/files/4638989/cluster.yaml.log)
* Operator's logs: [operator.log](https://github.com/rook/rook/files/4639004/operator.log)
* Crashing pod(s) logs: I'd love to, but there are none

**Environment**:
* OS (e.g. from /etc/os-release): Ubuntu 18.04.4 LTS
* Kernel (e.g. `uname -a`):  4.15.0-99-generic
* Cloud provider or hardware configuration: Bare metal with kubeadm, 5x Dell R720x with 9 drives each
* Rook version (use `rook version` inside of a Rook Pod):  v1.3.1
* Storage backend version (e.g. for ceph do `ceph -v`): 14.2.8
* Kubernetes version (use `kubectl version`): v1.18.2
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): N/A
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):
HEALTH_WARN 1 filesystem is degraded; 2 MDSs report slow metadata IOs; pauserd,pausewr,nodown,noout,nobackfill,norecover,noscrub,nodeep-scrub flag(s) set; Reduced data availability: 236 pgs inactive, 69 pgs peering; 79 pgs not deep-scrubbed in time; 18 daemons have recently crashed

**Comment by psarossy** [comment]
One more log, it seems to be related to the admin socket? (at the end of the log)

[rook-ceph-osd-11-77bc865f-tzfff.describe.log](https://github.com/rook/rook/files/4639054/rook-ceph-osd-11-77bc865f-tzfff.describe.log)

**Comment by psarossy** [comment]
A graceful power down and restart later, now I have 7 OSDs doing this...

Looking at the host processes, it looks like ceph-osd is never started in the container, then it tries to do the health check, which fails, then goes straight back to CrashLoopBackOff.

**Comment by psarossy** [comment]
So after 2 days of trying to figure out how to debug this, I found the source of the problem.
In some guides, for cluster reboots they also recommend for you to set "NOUP" on the OSDs.

Naturally this prevents the OSD from starting up properly:
```
$ kubectl logs -f -n rook-ceph rook-ceph-osd-14-cb8958458-dvfzc
2020-05-17 21:32:33.954 7fd5f721ea80 -1 Falling back to public interface
2020-05-17 21:33:50.267 7fd5f721ea80 -1 osd.14 21144 log_to_monitors {default=true}
2020-05-17 21:33:50.363 7fd5dd007700 -1 osd.14 21144 osdmap NOUP flag is set, waiting for it to clear
```

The problem is that the liveliness check only gives 10 seconds for the logs to get populated, but it takes ~30 seconds for that error message to show up, so the logs will not tell you this.

I ended up editing the deployment (stopped/scaled the operator first):
`$ kubectl -n rook-ceph edit deployment rook-ceph-osd-14`

And setting the livenessProbe to periodSeconds to 600 (but anything more than 1 minute would work), then when either just watch the OSD figure out that the NOUP has cleared on it's own or issue an `ceph osd unset noup` then delete/restart the pod, and "surprise", everything is back to normal.

**Essentially it takes up to a minute to get a new OSDMAP and flags from the monitors, the config only gives them 10 seconds and kills them actively.**

Gonna clean up this bug report and update it so it reflects the real bug.
